pytools/test/test_persistent_dict.py

```diff
@@ -432,9 +432,13 @@ def test_scalar_hashing() -> None:
     assert keyb(np.int32(1)) == keyb(np.int32(1))
     assert keyb(np.int32(2)) != keyb(np.int32(1))
     assert keyb(np.int64(1)) == keyb(np.int64(1))
-    assert keyb(1) == keyb(np.int64(1))
+    assert keyb(1) != keyb(np.int64(1))
```

Comment:
Wait, I thought the goal was to keep them the same?

Reply:
From your answer to [my question](https://github.com/inducer/pytools/pull/260#discussion_r1740209871) in #260, my interpretation was that they don't need to be the same. Changing the code like in this PR perhaps also addresses #262.